### ISSUE TITLE
Should work conditional expression without conditional statements

### ISSUE THREAD

**Issue by ajayojha** [rxweb/rxweb#114]
# Describe the bug
@boskee has highlighted the scenario in the issue of #111. conditional expression should work if the user puts without conditional statements.
```js
RxwebValidators.required({
                    conditionalExpression: (x) => {
                         return x.toggle;
                    }})
```

**Comment by yoan-asdrubal** [comment]
I having problen with conditionalExpresion inside required validator, I thought it was better to comment here than to open a new issue. My condition involves a formArray and required validator do not apply when i needed. Here an example.
https://stackblitz.com/edit/rxweb-validator-required-error?file=src%2Fapp%2Fapp.component.html

**Comment by ajayojha** [comment]
@yoan-asdrubal Thanks for reporting this issue.

This will be resolved in our next release, which will be available after 5 days. I tried with one other approach to give you a temporary solution, but that doesn't work :( .

**Comment by yoan-asdrubal** [comment]
Ok, thanks, in the meantime I solve it by dynamically handling the validations,  If possible let me know when the new version is available

**Comment by yoan-asdrubal** [comment]
Is there any way to implement some general validator? maybe combining this variant of conditionalExpression and specifying the error that I want to assign according to the condition.
maybe something like this
<pre>
RxwebValidators.custom({
  conditionalExpression: (x, y) => {
	 return x.genre  === 'Male' and y.age<18 ;
 },
  customError: 'maleNotAllowed',
  message:'Male under 18 years old not allowed'
 })
</pre>
I think it would be very useful.

**Comment by ajayojha** [comment]
> Ok, thanks, in the meantime I solve it by dynamically handling the validations, If possible let me know when the new version is available

@yoan-asdrubal  Will be released coming Friday or next week Wednesday.

**Comment by ajayojha** [comment]
> Is there any way to implement some general validator? maybe combining this variant of conditionalExpression and specifying the error that I want to assign according to the condition.
> maybe something like this
> 
> RxwebValidators.custom({
>   conditionalExpression: (x, y) => {
> 	 return x.genre  === 'Male' and y.age<18 ;
>  },
>   customError: 'maleNotAllowed',
>   message:'Male under 18 years old not allowed'
>  })
> I think it would be very useful.
Are you looking for custom validation or you want to show custom messages on particular FormControl?

**Comment by yoan-asdrubal** [comment]
I am looking for custom validation, I just put on example combining current config params and adding customError, its maybe something like messageKey, to differentiate between different validations of the same type. This example for instance, I dont know what validator to apply, and if there is any, I don't think I can identify the error that I am handling.

**Comment by yoan-asdrubal** [comment]
in the case of numerical validations, could the character by which I want to handle the decimals be added? either ',' or '.' This is probably the one that manages the browser by default, but in my case I took the ',' from the browser to apply the validation, maybe because of the language, and I couldn't use it.

**Comment by ajayojha** [comment]
> I am looking for custom validation, I just put on example combining current config params and adding customError, its maybe something like messageKey, to differentiate between different validations of the same type. This example for instance, I dont know what validator to apply, and if there is any, I don't think I can identify the error that I am handling.

@yoan-asdrubal You can use compose validator. Please refer this stackblitz url : https://stackblitz.com/edit/compose-validator-rxweb-for-custom-validation?file=src/app/compose-validators.component.html .

Use 'errorMessage' property for showing the error messages on the UI. With this, you don't need to put multiple if-else conditions on the HTML.

I hope the above link helps or else please let me know if you are looking something different than this.

**Comment by yoan-asdrubal** [comment]
oh that's great, in my case I have implemented my own validations, and I use them without problems in compose validation, only that in most cases it's basically the same, a condition and return a value that contains the error and probably the message. I thought it would be interesting to add an implementation that would allow you to have these validations easily, specifying the condition and the error to handle. In that case it would only be to declare the validation as they do at the moment all the validations of the library, and not repeat again and again the same code only to change condition and return. It would also be very useful for people who start with angular and do not understand how this topic works  of custom validations

**Comment by ajayojha** [comment]
> in the case of numerical validations, could the character by which I want to handle the decimals be added? either ',' or '.' This is probably the one that manages the browser by default, but in my case I took the ',' from the browser to apply the validation, maybe because of the language, and I couldn't use it.

@yoan-asdrubal Have you tried numeric validator with ```allowDecimal``` true.

**Comment by ajayojha** [comment]
> oh that's great, in my case I have implemented my own validations, and I use them without problems in compose validation, only that in most cases it's basically the same, a condition and return a value that contains the error and probably the message. I thought it would be interesting to add an implementation that would allow you to have these validations easily, specifying the condition and the error to handle. In that case it would only be to declare the validation as they do at the moment all the validations of the library, and not repeat again and again the same code only to change condition and return. It would also be very useful for people who start with angular and do not understand how this topic works  of custom validations

@yoan-asdrubal Looks Interesting :), But sorry to say I cannot summarize your context technically (in terms of developing a new feature). Would be great if you share little bit more info or rough code snippet.

**Comment by yoan-asdrubal** [comment]
> > in the case of numerical validations, could the character by which I want to handle the decimals be added? either ',' or '.' This is probably the one that manages the browser by default, but in my case I took the ',' from the browser to apply the validation, maybe because of the language, and I couldn't use it.
> 
> @yoan-asdrubal Have you tried numeric validator with `allowDecimal` true?

Yes, I tried.
<pre>
RxwebValidators.numeric(
{
    acceptValue: NumericValueType.PositiveNumber, allowDecimal: true
})
</pre>
but it does not work on my app, however if I try in other tab like stackblitz its work. my index.html have lang="en".

**Comment by yoan-asdrubal** [comment]
> > oh that's great, in my case I have implemented my own validations, and I use them without problems in compose validation, only that in most cases it's basically the same, a condition and return a value that contains the error and probably the message. I thought it would be interesting to add an implementation that would allow you to have these validations easily, specifying the condition and the error to handle. In that case it would only be to declare the validation as they do at the moment all the validations of the library, and not repeat again and again the same code only to change condition and return. It would also be very useful for people who start with angular and do not understand how this topic works  of custom validations
> 
> @yoan-asdrubal Looks Interesting :), But sorry to say I cannot summarize your context technically (in terms of developing a new feature). Would be great if you share little bit more info or rough code snippet.

I'll write some code example on stackblitz and I'll share to see this case. 
Thanks

**Comment by ajayojha** [comment]
> > > in the case of numerical validations, could the character by which I want to handle the decimals be added? either ',' or '.' This is probably the one that manages the browser by default, but in my case I took the ',' from the browser to apply the validation, maybe because of the language, and I couldn't use it.
> > 
> > 
> > @yoan-asdrubal Have you tried numeric validator with `allowDecimal` true?
> 
> Yes, I tried.
> 
> RxwebValidators.numeric(
> {
>     acceptValue: NumericValueType.PositiveNumber, allowDecimal: true
> })
> but it does not work on my app, however if I try in other tab like stackblitz its work. my index.html have lang="en".

@yoan-asdrubal Ok, I will check and let you know. Just for your info, you can join our [gitter channel](https://gitter.im/rxweb-project/rxweb) to discuss your minor integration related issues or feature request.

**Comment by ajayojha** [comment]
> > > in the case of numerical validations, could the character by which I want to handle the decimals be added? either ',' or '.' This is probably the one that manages the browser by default, but in my case I took the ',' from the browser to apply the validation, maybe because of the language, and I couldn't use it.
> > 
> > 
> > @yoan-asdrubal Have you tried numeric validator with `allowDecimal` true?
> 
> Yes, I tried.
> 
> RxwebValidators.numeric(
> {
>     acceptValue: NumericValueType.PositiveNumber, allowDecimal: true
> })
> but it does not work on my app, however if I try in other tab like stackblitz its work. my index.html have lang="en".
> I need number of decimal allowed too. In some cases I need to restrict the value of maximum number of decimal allowed

@yoan-asdrubal  I have checked with ```lang="en"``` and it's working fine in my machine. Would be great if you come on our [gitter channel](https://gitter.im/rxweb-project/rxweb) **(If possible, otherwise I am OK with this)** to drill down the problem.

**Comment by yoan-asdrubal** [comment]
Ok, I'll talk to you in gitter channel about this. 
Thanks

**Comment by ajayojha** [comment]
As @yoan-asdrubal issue has been resolved and available in the 1.9.2 release. Due to lots of change in the code if I am going to allow dev to write only ```t.toggle``` without any expression and the conditional expression works accordingly.

For a time being it's good if dev writes the code ```t.toggle == true```. Will fix later if the more request comes 😄 .
